src/x/net/http/errors.go

```diff
@@ -92,25 +92,30 @@ func WriteError(w http.ResponseWriter, err error, opts ...WriteErrorOption) {
 		fn(&o)
 	}
 
+	statusCode := getStatusCode(err)
+	if o.response == nil {
+		w.Header().Set(HeaderContentType, ContentTypeJSON)
+		w.WriteHeader(statusCode)
+		json.NewEncoder(w).Encode(ErrorResponse{Error: err.Error()})
+	} else {
+		w.WriteHeader(statusCode)
```

Comment:
Nit: looks like `statusCode` gets set in both branches, can probably move it outside?

Reply:
The order of `ResponseWriter` method calls is important. [From the docs](https://golang.org/pkg/net/http/#ResponseWriter):

> <...>
> Changing the header map after a call to WriteHeader (or Write) has no effect unless the modified headers are trailers.
> <...>
> If WriteHeader is not called explicitly, the first call to Write will trigger an implicit WriteHeader(http.StatusOK).
> <...>

It could be done, but either two `if`s on `o.response == nil` would be needed (one before `WriteHeader()` and one after), or introducing `defer`s for `Encode`/`Write` calls.